File: remind_me_be/RemindMe/RemindMe.Application/Requests/ToDos/CreateToDoRequest.cs
using MediatR;
using RemindMe.Domain.Entities;

namespace RemindMe.Application.Requests.ToDos
{
    public class CreateToDoRequest : IRequest<ToDo>
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
        public DateTime? CreationDate { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsFinished { get; set; }
        public int Difficulty { get; set; }
        public int OwnerId { get; set; }

    }
}

File: remind_me_fe/lib/models/dto_requests/create_todo_request.dart
class CreateToDoRequest {
  String title;
  String description;
  DateTime creationDate = DateTime.now();
  DateTime startDate;
  DateTime? endDate;
  bool isFinished;
  int difficulty;
  int ownerId;

  CreateToDoRequest({
    required this.title,
    required this.description,
    required this.creationDate,
    required this.startDate,
    this.endDate,
    required this.isFinished,
    required this.difficulty,
    required this.ownerId,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'description': description,
      'creationDate': creationDate.toIso8601String(),
      'startDate': startDate.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
      'isFinished': isFinished,
      'difficulty': difficulty,
      'ownerId': ownerId,
    };
  }
}
